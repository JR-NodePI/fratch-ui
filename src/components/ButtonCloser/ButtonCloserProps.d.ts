export interface ButtonCloserProps {
  className?: string;
  title?: string;
  onClick?: () => void;
}
