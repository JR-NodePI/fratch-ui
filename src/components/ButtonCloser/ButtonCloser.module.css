.button_closer {
  width: 42px;
  height: 42px;
  font-size: 24px;
  background-color: transparent;
  border: none;
  color: var(--ft-color-text);
  opacity: 0.5;
  position: absolute;
  right: 0;
  top: 0;
  border-radius: var(--ft-button-default-border-radius);
}
.button_closer:hover {
  opacity: 1;
}
