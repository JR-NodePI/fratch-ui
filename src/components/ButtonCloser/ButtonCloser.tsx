import { c } from '../../helpers/classNameHelpers';
import { IconClose } from '../Icons/Icons';
import { type ButtonCloserProps } from './ButtonCloserProps';

import styles from './ButtonCloser.module.css';

export default function ButtonCloser({
  className,
  onClick,
  title,
}: ButtonCloserProps): JSX.Element {
  return (
    <button
      className={c(styles.button_closer, className)}
      onClick={onClick}
      title={title}
    >
      <IconClose className={c(styles.icon)} />
      <i>{title}</i>
    </button>
  );
}
