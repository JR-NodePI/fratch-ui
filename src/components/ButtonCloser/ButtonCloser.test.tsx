import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, expect, it, vi } from 'vitest';

import ButtonCloser from './ButtonCloser';
import { type ButtonCloserProps } from './ButtonCloserProps';

describe('ButtonCloser', () => {
  const setup = (props: ButtonCloserProps = {}): ReturnType<typeof render> =>
    render(<ButtonCloser title="Test button" {...props} />);

  it('should render properly', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should render with className', () => {
    const { container } = setup({
      className: 'mock_test_class_name',
    });
    expect(container).toMatchSnapshot();
  });

  it('should perform onClick', async () => {
    const mockOnClick = vi.fn();

    setup({ title: 'mock-test-button', onClick: mockOnClick });

    await userEvent.click(
      screen.getByRole('button', { name: /mock-test-button/i })
    );

    expect(mockOnClick).toHaveBeenCalledTimes(1);
    expect(mockOnClick).toHaveBeenCalledWith();
  });
});
