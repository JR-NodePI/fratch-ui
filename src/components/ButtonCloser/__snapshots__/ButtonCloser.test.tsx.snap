// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button.tsx > should render properly 1`] = `
<div>
  <button
    class="_button_closer_8d9a3f"
    title="Test button"
  >
    <svg
      aria-hidden="true"
      class="_icon_2be8fc _primary_2be8fc _icon_8d9a3f"
      fill="none"
      viewBox="-1 -1 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
    <i>
      Test button
    </i>
  </button>
</div>
`;

exports[`Button.tsx > should render with className 1`] = `
<div>
  <button
    class="_button_closer_8d9a3f mock_test_class_name"
    title="Test button"
  >
    <svg
      aria-hidden="true"
      class="_icon_2be8fc _primary_2be8fc _icon_8d9a3f"
      fill="none"
      viewBox="-1 -1 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
    <i>
      Test button
    </i>
  </button>
</div>
`;
